### PR TITLE
More code excerpts for cookbook/persistence and cookbook/maintenance

### PR DESCRIPTION
**Pull request by khanhnwin**
**Please don't merge yet, this PR still requires some fixing on my end**

_Description of what this PR is changing or adding, and why:_ Migrating more snippets to code excerpts

_Issues fixed by this PR (if any):_

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: examples/development/tools/lib/hot-reload/after.dart [hunk 1]
```diff
@@ -30,28 +30,44 @@ final sampleTable = [
   Table(
     children: const [
       TableRow(
+<<<<<<< HEAD
```
Comment (domesticmouse) [inline examples/development/tools/lib/hot-reload/after.dart:33]: Are you meaning to check in diff markers?

File: examples/development/tools/lib/hot-reload/before.dart [hunk 2]
```diff
@@ -36,28 +36,44 @@ final sampleTable = [
   Table(
     children: const [
       TableRow(
+<<<<<<< HEAD
```
Comment (domesticmouse) [inline examples/development/tools/lib/hot-reload/before.dart:39]: diff marker

File: examples/ui/assets_and_images/analysis_options.yaml [hunk 3]
```diff
@@ -23,7 +23,11 @@ linter:
   # producing the lint.
   rules:
     # avoid_print: false  # Uncomment to disable the `avoid_print` rule
+<<<<<<< HEAD
```
Comment (domesticmouse) [inline examples/ui/assets_and_images/analysis_options.yaml:26]: diff marker

File: examples/ui/layout/adaptive_app_demos/lib/pages/focus_examples_page.dart [hunk 4]
```diff
@@ -1,6 +1,10 @@
 import 'package:adaptive_app_demos/global/device_type.dart';
 import 'package:flextras/flextras.dart';
 import 'package:flutter/material.dart';
+<<<<<<< HEAD
+=======
```
Comment (domesticmouse) [inline examples/ui/layout/adaptive_app_demos/lib/pages/focus_examples_page.dart:5]: diff marker

File: examples/ui/layout/adaptive_app_demos/lib/widgets/extra_widget_excerpts.dart [hunk 5]
```diff
@@ -1,3 +1,7 @@
+<<<<<<< HEAD
```
Comment (domesticmouse) [inline examples/ui/layout/adaptive_app_demos/lib/widgets/extra_widget_excerpts.dart:1]: diff marker

File: src/development/platform-integration/platform-channels.md [hunk 6]
```diff
@@ -761,11 +761,19 @@ languages are Objective-C, Java, Kotlin and Swift (via Objective-C interop).
 import 'package:pigeon/pigeon.dart';
 
 class SearchRequest {
+<<<<<<< HEAD
```
Comment (domesticmouse) [inline src/development/platform-integration/platform-channels.md:764]: diff marker

File: src/development/tools/hot-reload.md [hunk 7]
```diff
@@ -211,28 +211,44 @@ final sampleTable = [
   Table(
     children: const [
       TableRow(
+<<<<<<< HEAD
```
Comment (domesticmouse) [inline src/development/tools/hot-reload.md:214]: diff marker

**Review by domesticmouse — CHANGES_REQUESTED** [review]
There is a raft of diff markers in this PR. I'm guessing something went wrong somewhere

**Comment by domesticmouse** [comment]
Marked as draft, as @khanhnwin states in the header this is WIP

**Comment by khanhnwin** [comment]
Yep, something went wrong somewhere! I need to fix. Thank you for marking as draft @domesticmouse!

**Review by domesticmouse — APPROVED** [review]
This looks ready to go

**Comment by khanhnwin** [comment]
Closing this PR! Merged these code excerpts as part of #6785 Thanks @domesticmouse!

**Pull request closed without merging**